.github/workflows/spack_cpu_build.yaml

```diff
@@ -116,31 +120,38 @@ jobs:
             concretizer:
               reuse: dependencies
             config:
+              concretizer: clingo
+              source_cache: $SPACK_CACHE/source_cache
+              misc_cache: $SPACK_CACHE/misc_cache
+              build_stage: $SPACK_CACHE/build_stage
               install_tree:
                 root: /opt/spack
                 padded_length: False
             mirrors:
               local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
               spack: https://binaries.spack.io/develop
+            packages:
+              all:
+                require: "%gcc"
```

Comment:
For some reason, clang was getting picked up by default as the compiler, and this caused spack to fail horribly. This seems to be all that was necessary to fix the issue